server/src/main/java/ch/bifrost/server/impl/keyexchange/DHKeyExchangeServer.java

```diff
@@ -0,0 +1,47 @@
+package ch.bifrost.server.impl.keyexchange;
+
+import java.math.BigInteger;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.base.Optional;
+
+import ch.bifrost.core.api.datagram.DatagramEndpoint;
+import ch.bifrost.core.api.datagram.DatagramMessage;
+import ch.bifrost.core.api.keyexchange.DHKeyExchange;
+import ch.bifrost.core.api.keyexchange.IdKeyPair;
+import ch.bifrost.core.api.keyexchange.Key;
+import ch.bifrost.core.api.keyexchange.DHKeyExchangeRequestMessage;
+import ch.bifrost.core.api.keyexchange.DHKeyExchangeResponseMessage;
+import ch.bifrost.core.api.session.MultiplexingID;
+
+public class DHKeyExchangeServer implements KeyExchangeServer {
+
+	private DatagramEndpoint datagramEndpoint;
+
+	public DHKeyExchangeServer (DatagramEndpoint datagramEndpoint) {
+		this.datagramEndpoint = datagramEndpoint;
+	}
+
+	@Override
+	public Optional<IdKeyPair> get (MultiplexingID id, long timeout, TimeUnit unit) throws Exception {
```

Comment:
Well, right now the key exchange runs on a non-multiplexed datagram endpoint, while the payload server is multiplexed. The key exchange server knows about the id here only so that he can communicate to the client how to reach the payload server (it seems to me that the server and the client have to agree for which multiplexed port the key is to be used).

I guess we should discuss this point. Yet, in this PR I only wanted to add DH as a key exchange protocol. Changing the underlying datagram endpoint seems like a separate issue.